mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -421,6 +435,18 @@ object LinearRegression extends DefaultParamsReadable[LinearRegression] {
    */
   @Since("2.1.0")
   val MAX_FEATURES_FOR_NORMAL_SOLVER: Int = WeightedLeastSquares.MAX_NUM_FEATURES
+
+  /** String name for "auto". */
+  private[regression] val AUTO = "auto"
```

Comment:
I am probably forgetting, but I thought we decided against introducing constants in a similar situation last month? not sure if it's the same context